framework/test/jest/unit/specs/modules/chain/dpos/verifyBlockForger.spec.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { Dpos, Slots } = require('../../../../../../../src/modules/chain/dpos');
+const { constants } = require('../../../../utils');
+const { delegatePublicKeys } = require('./round_delegates');
+
+describe('dpos.verifyBlockForger()', () => {
+	describe('When DPoS returns correct delegate list', () => {
+		const stubs = {};
+		let dpos;
+		let slots;
+
+		beforeEach(() => {
+			// Arrange
+			stubs.storage = {
+				entities: {
+					RoundDelegates: {
+						getRoundDelegates: jest.fn().mockReturnValue(delegatePublicKeys),
+					},
+				},
+			};
+
+			stubs.logger = {
+				debug: jest.fn(),
+				log: jest.fn(),
+				error: jest.fn(),
+			};
+
+			slots = new Slots({
+				epochTime: constants.EPOCH_TIME,
+				interval: constants.BLOCK_TIME,
+				blocksPerRound: constants.ACTIVE_DELEGATES,
+			});
+
+			dpos = new Dpos({
+				slots,
+				...stubs,
+				activeDelegates: constants.ACTIVE_DELEGATES,
+			});
+		});
+
+		it('should resolve with "true" when block is forged by correct delegate', async () => {
+			// Arrange
+			const block = {
+				height: 302,
+				timestamp: 23450,
+				generatorPublicKey:
+					'6fb2e0882cd9d895e1e441b9f9be7f98e877aa0a16ae230ee5caceb7a1b896ae',
+			};
+
+			// Act
+			const result = await dpos.verifyBlockForger(block);
+
+			// Assert
+			expect(result).toBeTrue();
+		});
+
+		it('should throw error if block is forged by incorrect delegate', async () => {
+			// Arrange
+			const block = {
+				height: 302,
+				timestamp: 23450,
+				generatorPublicKey: 'xxx',
+			};
+
+			const expectedSlot = slots.getSlotNumber(block.timestamp);
+
+			// Act && Assert
+			const error = new Error(`Failed to verify slot: ${expectedSlot}`);
+			await expect(dpos.verifyBlockForger(block)).rejects.toEqual(error);
+		});
+	});
+
+	describe('When DPoS returns empty delegate list', () => {
+		const stubs = {};
+		let dpos;
+		let slots;
+
+		beforeEach(() => {
```

Comment:
actually you don't have to initialise everything from scratch here. You can simply update the return value of the mock in the arrange block of the test ? Example: https://github.com/LiskHQ/lisk-sdk/blob/6ec44cca9ed2d5fed60edcc65058834a350e6aa9/framework/test/jest/unit/specs/modules/chain/dpos/apply.spec.js#L307-L316

Reply:
Thanks for pointing that out!